iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ForExpressionImplTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.arm.parser.BicepParser;
+import org.sonar.iac.arm.parser.bicep.BicepLexicalGrammar;
+import org.sonar.iac.arm.parser.utils.Assertions;
+import org.sonar.iac.arm.tree.api.ArmTree;
+import org.sonar.iac.arm.tree.api.StringLiteral;
+import org.sonar.iac.arm.tree.api.bicep.ForExpression;
+import org.sonar.iac.arm.tree.api.bicep.ForVariableBlock;
+import org.sonar.iac.common.api.tree.TextTree;
+
+import static org.sonar.iac.common.testing.IacTestUtils.code;
+
+class ForExpressionImplTest {
+
+  BicepParser parser = BicepParser.create(BicepLexicalGrammar.FOR_EXPRESSION);
+
+  // TODO: SONARIAC-941 Add support for ifCondition
+  @Test
+  void shouldParseFunctionCall() {
+    Assertions.assertThat(BicepLexicalGrammar.FOR_EXPRESSION)
+      .matches("[for identifier123 in headerExpression:bodyExpression]")
+      .matches("[ for identifier123 in headerExpression : bodyExpression ]")
+      .matches("[for (itemIdentifier123,indexIdentifier123) in headerExpression:bodyExpression]")
```

Comment:
I would add a compact version to document how the parser behaves:
```
[for(itemIdentifier123,indexIdentifier123)in headerExpression:bodyExpression]
```
I guess it should match, if not we should check if this is a valid Bicep syntax.